pom.xml

```diff
@@ -465,6 +467,33 @@ under the License.
                     <doclint>none</doclint>
                 </configuration>
             </plugin>
+
+            <!-- make sure that flatten runs after shaded -->
```

Comment:
LGTM. minor suggestion: add comment to warn that this plugin takes effect for Maven version upper than 3.5.0.

Reply:
+1

Reply:
Done.